src/coreclr/jit/gentree.cpp

```diff
@@ -7167,7 +7171,7 @@ GenTree* Compiler::gtNewZeroConNode(var_types type)
 {
     GenTree* zero;
 
-    switch (type)
+    switch (genActualType(type))
```

Comment:
A lot of callers already call `gtNewZeroConNode(genActualType(type))`. As a matter of consistency, shouldn't we generally assume "actual" types everywhere, and callers who are dealing with full types call genActualType as necessary?

Reply:
I think it is an overall usability improvement for the method to accept small types. It's a bit of a "gotcha" that it didn't work for them in my opinion, and I personally spent a few write-build-SPMI cycles on fixing forgotten `genActualType`s around it (hence the change).

> A lot of callers already call `gtNewZeroConNode(genActualType(type))`

Hmm, I thought I deleted the obvious double `genActualType`s, but if some were missed, that will be fixed.